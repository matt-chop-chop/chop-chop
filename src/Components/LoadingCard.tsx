import { CircularProgress, Flex } from "@chakra-ui/react";

const LoadingCard = () => {
  return (
    <Flex
      alignItems="center"
      background="var(--chakra-colors-light-background)"
      border="2px solid var(--chakra-colors-light-emphasis)"
      borderRadius={4}
      height="100%"
      justifyContent="center"
      minHeight="250px"
      p={4}
      width="100%"
    >
      <CircularProgress
        color="var(--chakra-colors-light-surface)"
        isIndeterminate
        size={100}
      />
    </Flex>
  );
};

export default LoadingCard;
