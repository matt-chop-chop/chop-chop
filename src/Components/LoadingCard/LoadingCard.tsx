import { CircularProgress, Flex } from "@chakra-ui/react";
import React from "react";

export type LoadingCardProps = {
  showBorder?: boolean;
  mt?: string;
};

const LoadingCard = ({ showBorder = true, mt = "0px" }: LoadingCardProps) => {
  return (
    <Flex
      alignItems="center"
      background="var(--chakra-colors-light-background)"
      border={showBorder ? "2px solid var(--chakra-colors-light-text)" : ""}
      borderRadius={4}
      height="100%"
      justifyContent="center"
      minHeight="250px"
      mt={mt}
      p={4}
      width="100%"
    >
      <CircularProgress
        color="var(--chakra-colors-light-emphasis)"
        isIndeterminate
        size={100}
      />
    </Flex>
  );
};

export default LoadingCard;
