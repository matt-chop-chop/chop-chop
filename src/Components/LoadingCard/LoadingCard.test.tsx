import { shallow } from "enzyme";
import React from "react";
import LoadingCard, { LoadingCardProps } from "./LoadingCard";

const props: LoadingCardProps = {
  showBorder: false,
  mt: "10px",
};

describe("LoadingCard", () => {
  it("is exported from LoadingCard", () => {
    expect(LoadingCard).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(<LoadingCard {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders with default props ", () => {
    const wrapper = shallow(<LoadingCard />);

    expect(wrapper).toMatchSnapshot();
  });
});
