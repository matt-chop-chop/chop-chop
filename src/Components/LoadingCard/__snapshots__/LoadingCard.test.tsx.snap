// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingCard renders as expected  1`] = `
<Flex
  alignItems="center"
  background="var(--chakra-colors-light-background)"
  border=""
  borderRadius={4}
  height="100%"
  justifyContent="center"
  minHeight="250px"
  mt="10px"
  p={4}
  width="100%"
>
  <CircularProgress
    color="var(--chakra-colors-light-emphasis)"
    isIndeterminate={true}
    size={100}
  />
</Flex>
`;

exports[`LoadingCard renders with default props  1`] = `
<Flex
  alignItems="center"
  background="var(--chakra-colors-light-background)"
  border="2px solid var(--chakra-colors-light-text)"
  borderRadius={4}
  height="100%"
  justifyContent="center"
  minHeight="250px"
  mt="0px"
  p={4}
  width="100%"
>
  <CircularProgress
    color="var(--chakra-colors-light-emphasis)"
    isIndeterminate={true}
    size={100}
  />
</Flex>
`;
